azure/virtual-machine/variables.tf

```diff
@@ -0,0 +1,245 @@
+#########
+# Global
+#########
+
+variable "location" {
+  description = "The location of created resources."
+  type        = string
+  default     = "uksouth"
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group this module will use."
+  type        = string
+}
+
+variable "tags" {
+  description = "Tags applied to created resources."
+  type        = map(string)
+  default     = null
+}
+
+#########################
+# Virtual Machine Config
+#########################
+
+variable "name" {
+  description = "The name of the virtual machine."
+  type        = string
+}
+
+variable "os_type" {
+  description = "The OS type to use."
+  type        = string
+  default     = "Linux"
+
+  validation {
+    condition     = contains(["Windows", "Linux"], var.os_type)
+    error_message = "The OS type must be either Windows or Linux"
+  }
+}
+
+variable "computer_name" {
```

Comment:
Is it worth adding validation here? EG, less than 15 characters. 

I know this doesn't apply to Linux so perhaps it's worth adding a second variable called "Hostname" for Linux?

or maybe it's worth creating an object variable that contains both os_type, computer_name and a couple of other variables so that validation can be done such as "if linux, then validate"

Reply:
Leave this to the ARM API.

Reply:
Suggestion for a later update: Maybe look into using preconditions on the VM resource.

Reply:
Added into the board. https://github.com/orgs/ans-coe/projects/1?pane=issue&itemId=34170789